lib/spack/spack/config.py

```diff
@@ -539,13 +549,25 @@ def override(path_or_scope, value=None):
     an internal config scope for it and push/pop that scope.
 
     """
+    base_name = 'overrides-'
     if isinstance(path_or_scope, ConfigScope):
         overrides = path_or_scope
         config.push_scope(path_or_scope)
     else:
-        overrides = InternalConfigScope('overrides')
+        # Ensure the new override gets a unique scope name
+        current_overrides = [s.name for s in
+                             config.matching_scopes(r'^{0}'.format(base_name))]
+        num_overrides = len(current_overrides)
+        while True:
```

Comment:
Is this for the case where we have 3 overrides numbered 4,5,6?

Reply:
Yes, this is for nested overrides (e.g., `overrides_0`, `overrides_1`) etc.

I am following the naming convention (i.e., sequencing) I noticed with scopes entered on the command line.